java/src/client/game/GameManager.java

```diff
@@ -60,22 +82,24 @@ public void stopPoller() {
      * When the poller finds out the server has a new version it will call this function to update the client game
      * @param cm the new clientModel
      */
-    public void updateClientModels(ClientModel cm){
+    public void updateGameManager(ClientModel cm){
         mClientModel = cm;
+        mFacadeManager.update(cm);
     }
 
 
     /**
      * Grabs a "server" for us so we can talk with the real server
      * @return a server we can use
      */
+
     public IServer getServer() {
         return mServer;
     }
 
 
-    public static void setInstance(SessionManager sm) {
-        SessionManager.mInstance = sm;
+    public static void setInstance(GameManager sm) {
```

Comment:
Wait...what? Singletons are not to have their instance set publicly _ever_. Since the constructor is private, this function makes no sense.